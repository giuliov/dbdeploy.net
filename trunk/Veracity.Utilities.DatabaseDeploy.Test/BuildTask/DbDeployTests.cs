// --------------------------------------------------------------------------------------------------------------------
//  <copyright file="DbDeployTests.cs" company="Database Deploy 2">
//    Copyright (c) 2015 Database Deploy 2.  This code is licensed under the Microsoft Public License (MS-PL).  http://www.opensource.org/licenses/MS-PL.
//  </copyright>
//   <summary>
//  </summary>
// --------------------------------------------------------------------------------------------------------------------

namespace DatabaseDeploy.Test.BuildTask
{
    using System;

    using ApprovalTests;

    using DatabaseDeploy.Core;
    using DatabaseDeploy.Core.BuildTasks;
    using DatabaseDeploy.Core.Configuration;
    using DatabaseDeploy.Core.Database;
    using DatabaseDeploy.Core.IoC;
    using DatabaseDeploy.Core.Utilities;
    using DatabaseDeploy.Test.Utilities;

    using Microsoft.VisualStudio.TestTools.UnitTesting;

    using Moq;

    /// <summary>
    /// Tests of the DbDeploy Build Task.
    /// </summary>
    [TestClass]
    public class DbDeployTests : TestFixtureBase
    {
        /// <summary>
        /// Tests that execute sets up a run correctly.
        /// </summary>
        [TestMethod]
        public void ThatExecuteSetupsCorrectly()
        {
            MockEnvironmentProvider mockEnvironmentProvider = new MockEnvironmentProvider();
            mockEnvironmentProvider.SetCurrentDirectory(@"c:\DbDeploy");
            EnvironmentProvider.Current = mockEnvironmentProvider;

            try
            {
                IConfigurationService configurationService = new ConfigurationService();
                Container.RegisterInstance(configurationService);

                Mock<IDeploymentService> deploymentServiceMock = new Mock<IDeploymentService>(MockBehavior.Strict);
                deploymentServiceMock.Setup(d => d.BuildDeploymentScript()).Verifiable();
                Container.RegisterInstance(deploymentServiceMock.Object);

                DbDeploy deploy = new DbDeploy();
                deploy.ConnectionString = "Connection String";
                deploy.DatabaseType = "mssql";
                deploy.LastChangeToApply = 500;
                deploy.OutputFile = "Output File";
                deploy.Recursive = true;
                deploy.RootDirectory = "Root Directory";
                deploy.SearchPattern = "*.sql";
                deploy.UndoFile = "Undo File";

                deploy.Execute();

                Approvals.Verify(configurationService);

                deploy.DatabaseType = "ora";
                deploy.Execute();

                Assert.AreEqual(configurationService.DatabaseManagementSystem, DatabaseTypesEnum.Oracle);

                deploy.DatabaseType = "mysql";
                deploy.Execute();

                Assert.AreEqual(configurationService.DatabaseManagementSystem, DatabaseTypesEnum.MySql);

                deploy.DatabaseType = "BadDBType";
                bool expectedExceptionFound = false;
                try
                {
                    deploy.Execute();
                }
                catch (ArgumentException)
                {
                    expectedExceptionFound = true;
                }

                Assert.IsTrue(expectedExceptionFound, "Expected to receive an argument exception for a bad db type, but did not.");

                deploymentServiceMock.Verify(d => d.BuildDeploymentScript(), Times.Exactly(3));
            }
            finally
            {
                EnvironmentProvider.ResetToDefault();
            }
        }
    }
}