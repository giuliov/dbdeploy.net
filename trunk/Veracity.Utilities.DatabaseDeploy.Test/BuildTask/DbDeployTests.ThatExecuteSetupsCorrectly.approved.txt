ConnectionString: Connection String
DatabaseManagementSystem: SqlServer
LastChangeToApply: 500
OutputFile: Output File
Recursive: True
RootDirectory: C:\GitHub\dbdeploynet2\trunk\Veracity.Utilities.DatabaseDeploy.Test\bin\Debug\Root Directory
ScriptListFile: C:\GitHub\dbdeploynet2\trunk\Veracity.Utilities.DatabaseDeploy.Test\bin\Debug\Root Directory\ScriptList.txt
SearchPattern: *.sql
FileNamePattern: (\d+)(\s+)?(.+)?
UndoOutputFile: Undo File
